src/effgen.ml

```diff
@@ -200,6 +200,32 @@ module StaticGenerators = struct
                   (recgen (n / 2)) )
             ])
   ;;
+
+  let all_type_gen =
+    (* Generates all types available in Efftester *)
+    Gen.fix (fun recgen n ->
+        let base_types = [ Unit; Int; Float; Bool; String ] in
+        if n = 0
+        then Gen.oneofl base_types
+        else
+          Gen.frequency
+            [ (* Generate no alphas *)
+              (4, Gen.oneofl base_types);
+              (1, Gen.map (fun t -> Option t) (recgen (sqrt n)));
+              (1, Gen.map (fun t -> Ref t) (recgen (sqrt n)));
+              ( 1,
+                Gen.map
+                  (fun tuple_lst -> Tuple tuple_lst)
+                  (Gen.small_list (recgen (sqrt n))) );
```

Comment:
Shouldn't we guarantee that the size is `> 1` here, to create types that the rest of the system accepts? An easy way would be to just append two extra random-generated tuple parameters to the `small_list`.

Reply:
good catch!